paper-input-container.html

```diff
@@ -0,0 +1,275 @@
+      transform-origin: left top;
+      -webkit-transition: -webkit-transform 0.25s;
+      transition: transform 0.25s;
+
+      mixin(--paper-transition-easing);
+    }
+
+    .input-content.label-is-highlighted ::content label,
+    .input-content.label-is-highlighted ::content .paper-input-label {
+      color: var(--paper-input-container-focus-color);
+    }
+
+    .input-content.is-invalid ::content label,
+    .input-content.is-invalid ::content .paper-input-label {
+      color: var(--paper-input-container-invalid-color);
+    }
+
+    .input-content.label-is-hidden ::content label,
+    .input-content.label-is-hidden ::content .paper-input-label {
+      visibility: hidden;
+    }
+
+    .input-content ::content input,
+    .input-content ::content .paper-input-input {
+      position: relative; /* to make a stacking context */
+      outline: none;
+      color: var(--paper-input-container-input-color);
+
+      mixin(--paper-font-subhead);
+    }
+
+    .input-content ::content input {
+      font: inherit;
+      padding: 0;
+      width: 100%;
+      background: transparent;
+      border: none;
+
+      mixin(--paper-input-container-input);
+    }
+
+  </style>
+
+  <template>
+
+    <!-- <template is="x-if" if="[[!noLabelFloat]]"> -->
+      <div class="floated-label-placeholder">&nbsp;</div>
+    <!-- </template> -->
+
+    <div class$="[[_computeInputContentClass(noLabelFloat,focused,_inputHasContent,_inputIsInvalid)]]">
+      <content select=":not(.add-on)"></content>
+    </div>
+
+    <div class$="[[_computeUnderlineClass(focused,_inputIsInvalid)]]">
+      <div class="unfocused-line fit"></div>
+      <div class="focused-line fit"></div>
+    </div>
+
+    <div class="add-on-content">
+      <content select=".add-on"></content>
+    </div>
+
+  </template>
+
+</dom-module>
+
+<script>
+(function() {
+
+  Polymer({
+
+    is: 'paper-input-container',
+
+    enableCustomStyleProperties: true,
+
+    properties: {
+
+      /**
+       * Set to true to disable the floating label.
+       */
+      noLabelFloat: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * Set to true to auto-validate the input value.
+       */
+      autoValidate: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * True if the input has focus.
+       */
+      focused: {
+        readOnly: true,
+        type: Boolean,
+        value: false
+      },
+
+      _inputHasContent: {
+        type: Boolean,
+        value: false
+      },
+
+      _inputIsInvalid: {
+        type: Boolean,
+        value: false
+      }
+
+    },
+
+    listeners: {
+      'input': 'onInput'
+    },
+
+    attached: function() {
+      var input = Polymer.dom(this).querySelector('input,.paper-input-input');
+      this._inputHasContent = !!input.value;
+      this._inputIsInvalid = !input.validity.valid;
+
+      this._boundOnFocus = this.onFocus.bind(this);
+      this._boundOnBlur = this.onBlur.bind(this);
```

Comment:
It seems slightly wasteful to create these every time the element is `attached`. Can they be created in `properties` or `ready` instead?